test/WebJobs.Script.Tests/Description/FunctionAppValidationServiceTests.cs

```diff
@@ -178,7 +179,9 @@ public void GetOutdatedBundleWarningMessage_LogsWarning(string bundleId, string
             var options = new ExtensionBundleOptions { Id = bundleId };
             var env = new TestEnvironment();
             var config = new FunctionsHostingConfigOptions();
-            var manager = new ExtensionBundleManager(options, env, _loggerFactory, config);
+            var httpClientFactory = new Mock<IHttpClientFactory>();
+            httpClientFactory.Setup(f => f.CreateClient(It.IsAny<string>())).Returns(new HttpClient());
```

Comment:
We should add tests to cover various retry and failure scenarios, using a custom HttpMessageHandler to simulate different error conditions.

Reply:
Added Test cases not this test class but added.